### DISCUSSION TITLE
panic when getting an index that has none value

### DISCUSSION THREAD

**Discussion by marirs** [rust-lang/regex#783] [Q&A] [answered]
Error:
thread 'main' panicked at 'no group at index '1'', /Users/sgp/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-1.5.4/src/re_unicode.rs:1046:32

When the index 1; has None or no match, it is panic'ing. How can I handle this?
I have a regex of named captures, and not all might get results, since it covers multiple edge cases.
I have a multiline log file for which I use the captures_iter to match in the whole file, and when I try to get indexes from the `Captures` that dont have values, it panics
```
for caps in RE.captures_iter(s).take(3) {
        let x = caps.index(1));
        println!("{:?}", x)
    }
```

Appreciate if I can get a direction on this.
Thanks

**Comment by marirs** [comment, marked as answer]
Sorry - I think I got it done myself.
Instead of using .index 
I used the .get(1).map_or("", |m|m.as_str())

This helped me to put a default value of "" if the capture did not have a value.

Thanks !

**Reply by BurntSushi** [reply]
That looks good to me, yes. And note that it is a little weird to call `index` explicitly like that. You can instead use `caps[1]` for example.

The indexing notation is meant to be a convenience, just like for slices, where it panics if the index isn't valid. And just like slices, `Captures` provides a non-panicking `get` method (as you found).

**Reply by marirs** [reply]
Oh darn - how did I miss `caps[1]`; lol
Thanks so much!
